pennylane/ops/qubit/parametric_ops.py

```diff
@@ -35,6 +35,14 @@
 stack_last = functools.partial(qml.math.stack, axis=-1)
 
 
+def can_replace(x, y):
+    """
+    Convenience function that returns if x is close to y and if
+    x does not require grad
+    """
+    return qml.math.allclose(x, y) and not qml.math.requires_grad(x)
```

Comment:
```suggestion
    return (not qml.math.requires_grad(x)) and qml.math.allclose(x, y)
```
I was just reminded we need to make sure this logic is jit-compatible.  If we are tracing with an abstract jax tensor, then `qml.math.allclose` will cause `ConcretizationTypeError `.

Luckily, abstract types return `requires_grad(val)=True`, so we can exit early and not perform the `allclose` check.

If `requires_grad` didn't pick up on the abstract types, we would need to use `qml.math.is_abstract`.

You can test this out yourself with:
```
@jax.jit
@qml.qnode(qml.device('default.qubit.jax', wires=1), interface='jax')
@qml.simplify
def circuit(x, z):
    qml.adjoint(qml.RX(x, wires=0))
    qml.RZ(x, wires=0) ** 2
    return qml.expval(qml.PauliZ(0))

x = jax.numpy.array(0.1)
z = jax.numpy.array(2*jax.numpy.pi)

jax.grad(circuit)(x, z)
```

As long as you include the changes from #3017 at least.

Reply:
Right, that makes sense. I can add tests for abstract tensors then.